src/main/java/org/folio/search/utils/CallNumberUtils.java

```diff
@@ -144,6 +150,60 @@ public static Long getCallNumberAsLong(String callNumber, int firstPosition) {
     return callNumberToLong(callNumber, startVal, CN_MAX_CHARS - 1);
   }
 
+  public static List<CallNumberBrowseItem> excludeIrrelevantResultItems(String refinedCondition,
+                                                                        List<CallNumberBrowseItem> records) {
+    if (StringUtils.isBlank(refinedCondition)) {
+      return records;
+    }
+
+    records
+      .forEach(r -> r.getInstance().setItems(r.getInstance().getItems()
+        .stream()
+        .filter(i -> CallNumberType.fromId(i.getEffectiveCallNumberComponents().getTypeId())
+          .equals(CallNumberType.fromName(refinedCondition)))
+        .toList()));
+    records
+      .forEach(r -> r.getInstance().setHoldings(r.getInstance().getHoldings()
+        .stream()
+        .filter(h -> isInGivenType(refinedCondition, h.getCallNumber()))
+        .toList()));
+    var result = new ArrayList<>(records
+      .stream()
+      .filter(r -> r.getInstance().getItems()
+        .stream()
+        .anyMatch(i -> r.getFullCallNumber() == null
+          || i.getEffectiveCallNumberComponents().getCallNumber().equals(r.getFullCallNumber())))
+      .toList());
```

Comment:
What is the purpose of this filtration?
And the next one for holdings

Reply:
when one instance has two or more call numbers in different types, browsing one of those call numbers by type will bring all of the instance call numbers, because call number browse item includes instance information for each matching result, and instance will bring all of its call numbers.
All instance call numbers will be included in the browse result because of one of them .